scripts/Diffraction/isis_powder/polaris_routines/polaris_algs.py

```diff
@@ -139,17 +139,46 @@ def _merge_workspace_with_limits(focused_ws, q_lims):
     min_x = np.inf
     max_x = -np.inf
     num_x = -np.inf
-    for ws in focused_ws:
-        x_data = ws.dataX(0)
+    ws_max_range = 0
+    largest_range_spectrum = 0
+    for i in range(focused_ws.size()):
+        x_data = focused_ws[i].dataX(0)
         min_x = min(np.min(x_data), min_x)
         max_x = max(np.max(x_data), max_x)
         num_x = max(x_data.size, num_x)
-    width_x = (max_x-min_x)/num_x
-    binning = [min_x, width_x, max_x]
-    focused_ws = mantid.Rebin(InputWorkspace=focused_ws, Params=binning)
-    focused_ws_conjoined = mantid.ConjoinSpectra(InputWorkspaces=focused_ws)
+        ws_range = np.max(x_data)-np.min(x_data)
+        if ws_range > ws_max_range:
+            largest_range_spectrum = i + 1
+            ws_max_range = ws_range
+    focused_ws = mantid.Rebin(InputWorkspace=focused_ws, Params=[min_x, (max_x-min_x)/num_x, max_x])
```

Comment:
We need to check for `np.inf` before rebinning since that is now more likely if a bad dataset was fed in

For future devs we need a comment just briefly explaining that `min_x, (max_x-min_x)/num_x, max_x]` is linear binning (and why if possible)